Engine/OGF-Core/Core/Graphics/Framebuffer.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "Core/Core.h"
+#include "Core/Graphics/Texture.h"
+
+#include <GL/glew.h>
+
+#include <string>
+
+class Framebuffer
+{
+	friend class FramebufferCache;
+
+public:
+
+	enum class Attachment : GLenum {
+		COLOR_ATTACH = GL_COLOR_ATTACHMENT0,
+		DEPTH_ATTACH = GL_DEPTH_ATTACHMENT,
+		STENCIL_ATTACH = GL_STENCIL_ATTACHMENT
+	};
+
+	struct RenderBuffer {
+		uint32 id;
+		Texture::InternalFormat internalFormat;
+		Framebuffer::Attachment attachment;
+	};
+
+private:
+	uint32 _fbid;
+	RenderBuffer _renderbuffer;
+	Texture* _attachedTex;
+	std::string _name;
+
+public:
+	Framebuffer(const std::string& name);
+
+	// Prevent copying of framebuffers
+	Framebuffer(const Framebuffer&) = delete;
+	Framebuffer(Framebuffer&&) = default;
```

Comment:
The default move constructor won't work here, lets say you move the framebuffer, the old object will still delete the opengl framebuffer in its destructor